crates/task-impls/src/transactions.rs

```diff
@@ -425,9 +436,19 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> TransactionTaskState<TYPES,
         parent_comm: VidCommitment,
         view_number: TYPES::Time,
         parent_comm_sig: &<<TYPES as NodeType>::SignatureKey as SignatureKey>::PureAssembledSignatureType,
-    ) -> anyhow::Result<BuilderResponses<TYPES>> {
+    ) -> anyhow::Result<BuilderResponse<TYPES>> {
+        let version = match version(view_number, &self.decided_upgrade_certificate) {
+            Ok(v) if v.major == 0 && v.minor >= 1 && v.minor <= 3 => v,
```

Comment:
I think we should avoid having an upper bound here (and also compare `v` directly to a named version, rather than `v.major` and `v.minor`)